app/components/ui/button.tsx

```diff
@@ -0,0 +1,63 @@
+import type { ButtonHTMLAttributes } from 'react';
+
+type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
+  variant: 'primary' | 'secondary' | 'warm' | 'outlined';
+  buttonStyle: 'rounded' | 'fullyRounded';
+  size: 'small' | 'medium' | 'large';
+  children: React.ReactNode;
+};
+
+export function Button({ variant, buttonStyle, size, children, ...props }: ButtonProps) {
+  const baseClasses = 'w-full sm:w-fit me-2 mb-2 text-center font-medium focus:outline-none focus:ring-4';
+  let variantClasses = '';
+  let styleClasses = '';
+  let sizeClasses = '';
+
+  switch (variant) {
+    case 'primary':
+      variantClasses = 'bg-primary text-white focus:ring-gray-300 hover:bg-gray-900';
+      break;
+    case 'secondary':
+      variantClasses = 'bg-secondary focus:ring-teal-300 hover:bg-teal-200';
+      break;
+    case 'warm':
+      variantClasses = 'bg-warm text-white focus:ring-red-300 hover:bg-red-700';
+      break;
+    case 'outlined':
+      variantClasses = 'bg-white text-primary border border-gray-200 focus:ring-gray-300 hover:bg-gray-100';
+      break;
+    default:
+      variantClasses = 'bg-primary text-white focus:ring-gray-300 hover:bg-gray-900';
+  }
```

Comment:
I wounder would something like map make this shorter?

```const variantClassesMap: { [key: string]: string } = {
  'primary': 'bg-primary text-white focus:ring-gray-300 hover:bg-gray-900',
  'secondary': 'bg-secondary focus:ring-teal-300 hover:bg-teal-200',
  'warm': 'bg-warm text-white focus:ring-red-300 hover:bg-red-700',
  'outlined': 'bg-white text-primary border border-gray-200 focus:ring-gray-300 hover:bg-gray-100',
};```

and 
access like `variantClassesMap[variant ?? "primary"] ?? " ...default " `